src/internal.rs

```diff
@@ -424,7 +423,7 @@ impl Default for Rav1dFrameContext_task_thread_pending_tasks {
 
 #[repr(C)]
 pub(crate) struct Rav1dFrameContext_task_thread {
-    pub lock: pthread_mutex_t,
+    pub lock: Mutex<()>,
```

Comment:
What is this `Mutex` supposed to protect?  The change in this PR seems fine as an intermediate state, but we should figure out exactly what data this lock is protecting and put it inside.

Reply:
As far as I can tell there isn't data being locked here, it's to keep multiple threads from entering the critical section between `get_frame_progress` and updating the `f.sr_cur.progress` atomic. These accesses are all atomic but in this case need to be synchronized between separate atomic fields.

Reply:
Okay.  I guess I'm still not sure why exactly those accesses need to be synchronized together.  It's all good here, though, I'm just confused.